common/catalog-client/src/main/java/com/csgroup/s2pdgs/catalog/adapter/elasticsearch/ElasticsearchAssetCatalogClientManager.java

```diff
@@ -116,4 +122,31 @@ private SSLContext getSSlContext() {
             throw new RuntimeException(e);
         }
     }
+
+    private Mono<ElasticsearchAssetCatalogClient> getClientForCatalog(Catalog catalog) {
+        if (catalog instanceof ElasticsearchCatalog esCatalog) {
+            return Mono.fromCallable(() -> clientsByDefinition.compute(
+                esCatalog,
+                (k, v) -> Option.of(v).getOrElse(() -> clientFactory.apply(esCatalog))
+            ));
+        }
+        else {
+            return Mono.error(new RuntimeException("Unknown catalog type: %s".formatted(catalog)));
```

Comment:
Minor: I think that `IllegalArgumentException` is better than `RuntimeException` here
```suggestion
            return Mono.error(new IllegalArgumentException("Unknown catalog type: %s".formatted(catalog)));
```